src/Runner.Common/ProcessChannel.cs

```diff
@@ -96,5 +98,21 @@ private void Dispose(bool disposing)
                 _outClient?.Dispose();
             }
         }
+
+        //  Create a deterministic hash for a string to ensure that the message is not malformed on delivery
+        private static string GetMessageHash(string message)
```

Comment:
There is an `IOUtil.GetPathHash()` that doing the exact thing, you might want to just reuse that.
Maybe rename that method to `IOUtil.GetSha256Hash()`